### PR TITLE
feat(ci): Add release workflow

### PR DESCRIPTION
**Pull request by phated**
# Related issue(s)

(If it does not already exist, first create a GitHub issue that describes the problem this Pull Request (PR) solves before creating the PR and link it here.)

Prerequisite to adding automated releases as per #19 (Need to discuss how much we actually want automated).

# Description

This adds Release Please to the project. The setup is different from the noir repo because we want to track changes to each package independently. We do this with the `cargo-workspace` plugin, but we also want all versions to stay in-sync so we use the `linked-versions` plugin to ensure that.

We also want to track all changes in a changelog in the project root so we have a "simple" releaser pointing at the root directory.

You can see an example of the Release Please PR at https://github.com/phated/acvm/pull/1

## Summary of changes

(Describe the changes in this PR. Point out breaking changes if any.)

## Dependency additions / changes

(If applicable.)

## Test additions / changes

(If applicable.)

# Checklist

- [x] I have tested the changes locally.
- [x] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` with default settings.
- [x] I have [linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) this PR to the issue(s) that it resolves.
- [x] I have reviewed the changes on GitHub, line by line.
- [x] I have ensured all changes are covered in the description.

# Additional context

(If applicable.)

### CODE REVIEW

File: .release-please-manifest.json [hunk 1]
```diff
@@ -0,0 +1 @@
+{".":"0.4.1","acir":"0.4.1","acir_field":"0.4.1","acvm":"0.4.1","stdlib":"0.4.1"}
```
Comment (kevaundray) [inline .release-please-manifest.json:1]: This gets updated by release-please when a breaking change is added via a commit?

Comment (phated) [reply .release-please-manifest.json:1]: It's updated by release-please for any version bumps. 🎉

Comment (kevaundray) [reply .release-please-manifest.json:1]: Oh nice!

**Review by kevaundray — APPROVED** [review]

**Comment by phated** [comment]
I just realized I wanted to double-check your bootstrap sha. 1 second.

**Comment by phated** [comment]
It looks like 97149adb99f53e1f78de6d7ad938f4af262483e6 is the release of 0.4.1 🎉

**Review by kevaundray — APPROVED** [review]

**Pull request merged**